diff --git a/lib/index.cjs b/lib/index.cjs
index 13872b5626beaa7c5d4ab6009beff5b12bcb0b8b..ad595ecd1cfa7df731d9f90dc315157f353d2ba0 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -843,7 +843,7 @@ var Satori = class extends import_cordis5.Service {
     (0, import_cosmokit5.defineProperty)(this.bots, import_cordis5.Service.tracker, {});
     const self = this;
     ctx.on("http/file", async function(_url, options) {
-      const url = new URL(_url);
+      const url = new URL(this.resolveURL(_url, this.config));
       if (url.protocol !== "internal:") return;
       const { status, body, headers } = await self.handleInternalRoute("GET", url);
       if (status >= 400) throw new Error(`Failed to fetch ${_url}, status code: ${status}`);
